### DISCUSSION TITLE
Upgrade / move stations and depots

### DISCUSSION THREAD

**Discussion by Mattaneua51** [voxeltycoon/issues#674] [UI&UX]
I would absolutely love the ability to upgrade and/or move a station, depot, garage, etc as soon as you have the next higher tier available instead of having to destroy it even if it still had a cost associated with the project. right now to put in a garage 2 you have to completely remove old construction but that seems inefficient especially business wise where as it could be you click upgrade and it shows you the new blueprint. It could technically also move the routes associated with it for ease of logistics.

**Comment by imcasper** [comment]
This is especially important when it comes time to improve the station. Now it's not only expensive (let's justify it with game mechanics), but also inconvenient.

**Comment by Arsaell** [comment]
And I would love the ability to extend a train station (add lanes, or length) after initial construction. 
Provided, of course, there's enought free space around!
